tests/utils_tests/test_archive.py

```diff
@@ -42,6 +44,21 @@ def test_extract_function(self):
         extract(self.archive_path, self.tmpdir)
         self.check_files(self.tmpdir)
 
+    @unittest.skipIf(
+        sys.platform == 'win32',
+        'Python on Windows has a limited os.chmod() support.',
+    )
+    def test_extract_file_permissions(self):
+        """
+        Make sure permissions are kept after extraction from an archive.
+        In this case, executable permission is used for testing.
+        """
+        extract(self.archive_path, self.tmpdir)
+        filepath = os.path.join(self.tmpdir, '1')
```

Comment:
What change did yo make the archives? is `1` added or changed? It would be self-documenting, for example, if the file name was something like "executable".